external/fv3fit/tests/reservoir/test_reservoir_computing_model.py

```diff
@@ -215,11 +223,175 @@ def test_hybrid_gives_different_results():
 
     for i in range(10):
         predictions.append(predictor.predict())
-        predictions_hybrid.append(hybrid_predictor.predict(predictions_hybrid[-1]))
+        predictions_hybrid.append(
+            hybrid_predictor.predict(
+                predictions_hybrid[-1], imperfect_model=imperfect_model
+            )
+        )
 
     predictions = np.array(predictions)
     predictions_hybrid = np.array(predictions_hybrid)
     assert predictions.shape == predictions_hybrid.shape
     np.testing.assert_raises(
         AssertionError, np.testing.assert_array_equal, predictions, predictions_hybrid
     )
+
+
+def create_domain_predictor(type, domain_size, subdomain_size, subdomain_overlap):
+
+    n_subdomains = domain_size // subdomain_size
+
+    hyperparameters = ReservoirHyperparameters(
+        input_size=subdomain_size + 2 * subdomain_overlap,
+        state_size=20,
+        adjacency_matrix_sparsity=0.0,
+        spectral_radius=1.0,
+        input_coupling_sparsity=0,
+    )
+    subdomain_predictors = []
+    for i in range(n_subdomains):
+        reservoir = Reservoir(hyperparameters)
+        readout = ReservoirComputingReadout(
+            linear_regressor=MultiOutputMeanRegressor(n_outputs=subdomain_size),
+            square_half_hidden_state=False,
+        )
+        if type == "reservoir_only":
+            subdomain_predictors.append(
+                ReservoirComputingModel(reservoir=reservoir, readout=readout,)
+            )
+
+        elif type == "hybrid":
+            subdomain_predictors.append(
+                HybridReservoirComputingModel(reservoir=reservoir, readout=readout,)
+            )
+
+    if type == "reservoir_only":
+        return ReservoirOnlyDomainPredictor(
+            subdomain_predictors=subdomain_predictors,
+            subdomain_config=SubdomainConfig(
+                size=subdomain_size, overlap=subdomain_overlap
+            ),
+        )
+    elif type == "hybrid":
+        return HybridDomainPredictor(
+            subdomain_predictors=subdomain_predictors,
+            subdomain_config=SubdomainConfig(
+                size=subdomain_size, overlap=subdomain_overlap
+            ),
+        )
+    else:
+        raise ValueError("domain predictor must be of type 'reservoir_only' or 'hybrid")
+
+
+def test_DomainPredictor_synchronize_updates_states():
+    domain_size = 6
+    subdomain_size = 2
+    subdomain_overlap = 1
+    n_subdomains = domain_size // subdomain_size
+    domain_predictor = create_domain_predictor(
+        "reservoir_only", domain_size, subdomain_size, subdomain_overlap
+    )
+    burnin = np.arange(domain_size * 10).reshape(10, domain_size)
+
+    states_init = [
+        domain_predictor.subdomain_predictors[i].reservoir.state
+        for i in range(n_subdomains)
+    ]
```

Comment:
Maybe this as a method on the `DomainPredictor` class?  The pattern is used fairly often at least in the tests.